fastlane/lib/fastlane/actions/device_grid/README.md

```diff
@@ -0,0 +1,106 @@
+# fastlane danger Device Grid
+
+Ever dream of testing your app straight from a pull request? Well now you can! With [fastlane](https://fastlane.tools), [danger](https://github.com/danger/danger) and [appetize.io](https://appetize.io/), you can stream your latest changes right from the browser.
+
+No more manually installing and testing your app just to review a PR.
+
+![assets/GridExampleScreenshot.png](assets/GridExampleScreenshot.png)
+
+## Requirements
+
+- [fastlane](https://fastlane.tools)
+- [danger](https://github.com/danger/danger)
+- [appetize.io](https://appetize.io/) account
+- A Continuous Integration system
+
+## Getting started
+
+### Install fastlane and danger
+
+Create a `Gemfile` in your project's directory with the following content
+
+```ruby
+gem "fastlane"
+gem "danger"
+```
+
+and run
+
+```
+bundle install
+```
+
+### Setup `fastlane`
+
+Skip this step if you're already using `fastlane` (which you should)
+
+```
+fastlane init
+```
+
+### Setup `danger`
+
+```
+danger init
+```
+
+Follow the `danger` guide to authenticate with GitHub
+
+### Configure [fastlane](https://fastlane.tools)
+
+Edit `fastlane/Fastfile`. Feel free to remove the auto-generated lanes. Add the following lane:
+
+```ruby
+desc "Build your app and upload it to Appetize to stream it in your browser"
+lane :upload_to_appetize do
+  import_from_git(url: "https://github.com/fastlane/fastlane",
+                 path: "fastlane/lib/fastlane/actions/device_grid/AppetizeFastfile")
+
+  build_and_upload_appetize(
+    xcodebuild: {
+      workspace: "YourApp.xcworkspace",
+      scheme: "YourScheme"
+    }
+  )
+end
+```
+
+Make sure to fill in your actual workspace and scheme, or replace those 2 parameters with `project: "YourApp.xcworkspace"`
+
+### Configure `danger`
+
+Edit `Dangerfile` and replace the content with
+
+```ruby
+puts "Running fastlane to generate and upload an ipa file..."
+puts `fastlane upload_to_appetize` # this will generate and upload your ipa file
+
+import "https://raw.githubusercontent.com/fastlane/fastlane/master/fastlane/lib/fastlane/actions/device_grid/device_grid.rb"
```

Comment:
Hmm, same as above with the `import_from_git`.  Another downside to this is that we'll be cementing our current git directory structure in a way we can't change.  If we'd done this before we wouldn't be able to move to the monorepo.